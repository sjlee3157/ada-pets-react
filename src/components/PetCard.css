&::-webkit-scrollbar {
    display: none;
}

@media(min-width:768px) {
    width:80%;
    white-space:nowrap;
    margin-top: 0;
}

.pet-card {
  margin: 60px 30px 0px 30px;
  display: inline-block;

  .pet-card--body {
    white-space: normal;
  }
}


.pet-card--select-pet-btn {
  margin-left: 15px;
}

.pet-card--close-btn {
  float: right;
}
