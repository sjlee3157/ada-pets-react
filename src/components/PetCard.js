import React from 'react';
import PropTypes from 'prop-types';

import 'bootstrap/dist/css/bootstrap.min.css';
import './PetCard.css';

import speciesEmoji from '../speciesEmoji';


const PetCard = (props) => {
  const { id, name, species, about, location, images } = props;
  return (
    <div className="card pet-card">
      <section className="card--header">

      { speciesEmoji(species) } {id} - {name}
        <button
          className="btn btn-primary pet-card--select-pet-btn"
          onClick={ () => props.onSelectPetCallback({ id, name, species, about, location, images }) }
          >
            Select
        </button>
        <button
          type="button"
          className="btn btn-danger pet-card--close-btn"
          aria-label="Close"
          onClick={ () => props.onRemovePetCallback(id) }
        >
          Remove
        </button>
      </section>
      <section className="pet-card--body">
        { about.length > 128 ? `${about.substring(0, 128)}...` : about}
      </section>
      <section className="pet-card--footer text-muted">
        {location}
      </section>
    </div>
  );
};

PetCard.propTypes = {
  id: PropTypes.number.isRequired,
  images: PropTypes.array.isRequired,
  name: PropTypes.string.isRequired,
  species: PropTypes.string.isRequired,
  about: PropTypes.string,
  location: PropTypes.string,
  onSelectPetCallback: PropTypes.func.isRequired,
  onRemovePetCallback: PropTypes.func.isRequired
}

export default PetCard;
