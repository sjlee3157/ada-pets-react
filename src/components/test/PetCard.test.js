import React from 'react';
import PetCard from '../PetCard';
import { shallow } from 'enzyme';

describe('petcard', () => {
  
  it('will match the PetCard Snapshot', () => {
    const wrapper = shallow( <PetCard 
      species="dog" 
      name="stinker"
      id={1}
      about="stinky"
      images={["pic"]}
      deletePetCallback={() => {} }
      selectPetCallback={() => {}}
    /> );

    expect(wrapper).toMatchSnapshot();
  });

});