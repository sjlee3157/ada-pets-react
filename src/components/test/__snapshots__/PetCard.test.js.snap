// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`petcard will match the PetCard Snapshot 1`] = `
<div
  className="card pet-card"
>
  <h1>
    Pet Card!
  </h1>
  <section
    className="pet-card--header"
  >
    <button
      aria-label="Close"
      className="close pet-card--close-btn"
      onClick={[Function]}
      type="button"
    >
      <span
        aria-hidden="true"
      >
        ×
      </span>
    </button>
    🐕
     
    1
     - 
    stinker
    <button
      className="btn btn-primary pet-card--select-pet-btn"
      onClick={[Function]}
    >
      Select
    </button>
  </section>
  <section
    className="pet-card--body"
  >
    stinky
  </section>
  <section
    className="pet-card--footer text-muted"
  />
</div>
`;
